### PR TITLE
Add more information when throwing errors in discoverying 3rd party resources

### PR DESCRIPTION
**Pull request by janetkuo**
#26425

[Analytics]()

### CODE REVIEW

File: pkg/kubectl/cmd/util/factory.go [hunk 1]
```diff
@@ -250,17 +250,23 @@ func NewFactory(optionalClientConfig clientcmd.ClientConfig) *Factory {
 		// have been dynamically added to the apiserver
 		Object: func(discoverDynamicAPIs bool) (meta.RESTMapper, runtime.ObjectTyper) {
 			cfg, err := clientConfig.ClientConfig()
-			CheckErr(err)
+			if err != nil {
+				CheckErr(fmt.Errorf("failed to get client config: %v", err))
```
Comment (caesarxuchao) [inline pkg/kubectl/cmd/util/factory.go]: How about plumbing the prefix to the `checkErr` function, then you won't need the `err != nil` checks.

**Comment by janetkuo** [comment]
@caesarxuchao ptal

**Comment by caesarxuchao** [comment]
LGTM, but it needs rebase.

**Comment by roberthbailey** [comment]
Should this be cherry picked into the 1.3 branch as well?

**Comment by caesarxuchao** [comment]
The bug is fixed in another PR. This one adds more log. I don't think it needs to be cherry-picked.

**Comment by roberthbailey** [comment]
Thanks @caesarxuchao. I was asking b/c it's at the top of the merge queue interspersed with the 1.3 changes but wasn't marked for 1.3.

**Comment by eparis** [comment]
i'm going to remove the p1 label since it doesn't seem like a high priority (which will address @roberthbailey comment about the queue position)

**Pull request merged**